### PR TITLE
[FIX] Move complement.hpp

### PR DESCRIPTION
**Pull request by eaasna**
Resolves https://github.com/seqan/product_backlog/issues/336

- moved `complement.hpp` and the corresponding snippet to `seqan3/alphabet/views/`
- moved `complement_test`
- deprecated old header
- updated includes

### CODE REVIEW

File: include/seqan3/alphabet/views/complement.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * | seqan3::const_iterable_range     |                                       | *preserved*                                                     |
+ * |                                  |                                       |                                                                 |
+ * | std::ranges::range_reference_t   | seqan3::nucleotide_alphabet           | std::remove_reference_t<std::ranges::range_reference_t<urng_t>> |
+ *
+ * See the \link views views submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * ### Example
+ *
+ * \include test/snippet/alphabet/views/complement.cpp
+ * \hideinitializer
```
Comment (marehr) [inline include/seqan3/alphabet/views/complement.hpp:66]:
```suggestion
 * \hideinitializer
 *
 * \stableapi{Since version 3.1.}
```

**Review by MitraDarja — APPROVED** [review]
LGTM, but you touch a couple of files where seqan3/std-includes are not following our new standard (seqan3/std is supposed to go where std would go), it would be nice if you could fix that. :)

**Review by eseiler — CHANGES_REQUESTED** [review]
What Mitra said :)

And you'll need to rebase / resolve conflicts

**Comment by eaasna** [comment]
Okay thanks I will place the std includes separately :) 

Should I also move `test/unit/range/views/complement_test.cpp` to `test/unit/alphabet/views/complement_test.cpp`? @eseiler

**Comment by eseiler** [comment]
> Okay thanks I will place the std includes separately :)
> 
> Should I also move `test/unit/range/views/complement_test.cpp` to `test/unit/alphabet/views/complement_test.cpp`? @eseiler

Yes, please :)

**Comment by eaasna** [comment]
I moved the test, created a new CMakeLists.txt and rebased. Please check again @eseiler

File: test/unit/range/views/get_test.cpp [hunk 2]
```diff
@@ -10,16 +10,16 @@
 #include <gtest/gtest.h>
 
 #include <range/v3/view/zip.hpp>
+#include <seqan3/std/ranges>
```
Comment (eseiler) [inline test/unit/range/views/get_test.cpp:13]: Can you move this up to the `iostream` include?

File: include/seqan3/range/views/all.hpp [hunk 3]
```diff
@@ -13,7 +13,8 @@
 #pragma once
 
 #include <seqan3/alphabet/views/char_to.hpp>
-#include <seqan3/range/views/complement.hpp>
+#include <seqan3/alphabet/views/complement.hpp>
+#include <seqan3/range/views/convert.hpp>
 #include <seqan3/range/views/deep.hpp>
```
Comment (eseiler) [inline include/seqan3/range/views/all.hpp:18]:
```suggestion
#include <seqan3/alphabet/views/complement.hpp>
#include <seqan3/range/views/deep.hpp>
```

**Review by eseiler — CHANGES_REQUESTED** [review]

**Review by eseiler — APPROVED** [review]

File: include/seqan3/alphabet/views/complement.hpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides seqan3::views::complement.
+ */
+
+#pragma once
+
+#include <seqan3/std/ranges>
+
+#include <seqan3/alphabet/nucleotide/concept.hpp>
+#include <seqan3/range/views/deep.hpp>
```
Comment (eseiler) [inline include/seqan3/alphabet/views/complement.hpp:18]:
```suggestion
#include <seqan3/utility/views/deep.hpp>
```

File: test/unit/search/views/kmer_hash_test.cpp [hunk 5]
```diff
@@ -9,10 +9,11 @@
 #include <list>
 #include <type_traits>
 
+
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
```
Comment (eseiler) [inline test/unit/search/views/kmer_hash_test.cpp:13]:
```suggestion

#include <seqan3/alphabet/nucleotide/dna4.hpp>
```

**Pull request merged**